### ISSUE TITLE
Flaky Set test

### ISSUE THREAD

**Issue by ljharb** [paulmillr/es6-shim#268]
`Collections Set has an iterator that works with Array.from works with the full set:` will sometimes pass, and sometimes fail with:
`AssertionError: expected [ false, NaN,  ] to deeply equal [ 1, NaN, false ]`

I think this might be because the iterator that `Set` (and possibly `Map`) returns shares it's state across instances, rather than always being a completely new instance.

@cscott, if you have any time, could you take a look?

**Comment by ljharb** [comment]
This just happened again. https://travis-ci.org/ljharb/es6-shim/jobs/57920037
```shell
 1) Collections Set has an iterator that works with Array.from works with the full set:
      AssertionError: expected [ false, NaN,  ] to deeply equal [ 1, NaN, false ]
      + expected - actual
       [
      +  1
      +  NaN
         false
      -  NaN
      -  [undefined]
       ]
      
      at Context.<anonymous> (test/collections.js:801:36)
```

**Comment by ljharb** [comment]
Turns out chai and assert have a "deep equal" bug with NaN, at least in node 0.10. Fixed in e96091f9ec86aa7ba5178c94b6b34426496bde1d.
